packages/cli/src/commands/addons/wait.ts

```diff
@@ -0,0 +1,72 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {Args, ux} from '@oclif/core'
+import * as Heroku from '@heroku-cli/schema'
+import {resolveAddon} from '../../lib/addons/resolve'
+import {waitForAddonProvisioning, waitForAddonDeprovisioning} from '../../lib/addons/addons_wait'
+import notify from '../../lib/notify'
+
+export default class Wait extends Command {
+    static topic = 'addons';
+    static description = 'show provisioning status of the add-ons on the app';
+    static flags = {
+      'wait-interval': flags.string({description: 'how frequently to poll in seconds'}),
+      app: flags.app(),
+    };
+
+    static args = {
+      addon: Args.string(),
+    };
+
+    public async run(): Promise<void> {
+      const {flags, args} = await this.parse(Wait)
+      let addonsToWaitFor
+      if (args.addon) {
+        addonsToWaitFor = [await resolveAddon(this.heroku, flags.app, args.addon)]
+      } else if (flags.app) {
+        const {body: addons} = await this.heroku.get<Heroku.AddOn[]>(`/apps/${flags.app}/addons`)
+        addonsToWaitFor = addons
+      } else {
+        const {body: addons} = await this.heroku.get<Heroku.AddOn[]>('/addons')
+        addonsToWaitFor = addons
+      }
+
+      addonsToWaitFor = addonsToWaitFor.filter(addon => addon.state === 'provisioning' || addon.state === 'deprovisioning')
+      let interval = Number.parseInt(flags['wait-interval'] || '', 10)
+      if (!interval || interval < 0) {
+        interval = 5
+      }
+
+      for (const addon of addonsToWaitFor) {
+        const startTime = new Date()
+        const addonName = addon.name
+        if (addon.state === 'provisioning') {
+          let addonResponse
+          try {
+            // eslint-disable-next-line no-await-in-loop
```

Comment:
Is the intention here to fail all remaining when at least 1 addon fails to provision? If not, consider the use of `Promise.allSettled()` here so you can still take advantage of parallelization. e.g.
```
const results = await Promise.allSettled(addonsToWaitFor.map(addon => waitForAddonProvisioning(this.heroku, addon, interval))

for (let i = 0; i < results.length; i++) {
  const result = results[0];
  const {name: addonName} = addonsToWaitFor[i]
  if (result.status === 'rejected') {
    notify(`heroku addons:wait ${addonName}`, 'Add-on failed to provision', false)
    continue;
  }
  // ... The rest of your logic
}
```
If needed, for addons that are deprovisioning with a timeout, `Promise.race()` can be returned.

Reply:
@justinwilaby The `wait` commands are useful when you're scripting workflows based on the CLI and you need to verify that no prior pending operation failed (in this case `addons:create` or `addons:destroy`) before continuing with the flow of operations. So, ultimately you won't be waiting on the remaining add-ons if at least one of them fails. There will be no performance improvement with parallelization as this will always wait on the slowest of all add-ons to provision or fail and the same will occur with `allSettled`.

Reply:
Refactoring this to be parallelized is appealing, but I'm concerned about time spent to make sure we haven't introduced a problem. The focus right now is just on converting the command to oclif. Perhaps we can revisit this in Phase IV cleanup.